### PR TITLE
feat: Websocket support

### PR DESCRIPTION
**Pull request by tjholm**
Adds websocket support to the node SDK.

### CODE REVIEW

File: src/api/websocket/v0/websocket.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+// Storage client singleton
+let WEBSOCKET = undefined;
+
+/**
+ * Websocket API client.
+ *
+ * @returns a Storage API client.
+ * @example
+ * ```typescript
+ * import { storage } from "@nitric/sdk";
```
Comment (davemooreuws) [inline src/api/websocket/v0/websocket.ts:85]: Change these comments

File: src/api/websocket/v0/websocket.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+import { WebsocketServiceClient } from '@nitric/api/proto/websocket/v1/websocket_grpc_pb';
+import {
+  WebsocketSendRequest,
+  WebsocketCloseRequest,
+} from '@nitric/api/proto/websocket/v1/websocket_pb';
+import * as grpc from '@grpc/grpc-js';
+import { fromGrpcError } from '../../errors';
+
+/**
+ * Nitric storage client, facilitates writing and reading from blob storage (buckets).
```
Comment (davemooreuws) [inline src/api/websocket/v0/websocket.ts:24]: Change these comments

File: src/resources/websocket.test.ts [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+      });
+
+      it('Should return the same resource object', () => {
+        expect(websocketResource === secondWebsocket).toEqual(true);
+      });
+    });
+  });
+});
+
+// describe('subscription', () => {
```
Comment (davemooreuws) [inline src/resources/websocket.test.ts:132]: Remove this commented out code

File: src/resources/websocket.ts [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+    this.faas.websocket(...middleware);
+  }
+
+  private async start(): Promise<void> {
+    return this.faas.start();
+  }
+}
+
+/**
+ * Cloud storage bucket resource for large file storage.
```
Comment (davemooreuws) [inline src/resources/websocket.ts:68]: Change these comments

File: src/resources/websocket.ts [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+export class WebsocketResource extends Base<any> {
+  private readonly wsClient: WsClient;
+
+  constructor(name: string) {
+    super(name);
+    this.wsClient = wsClient();
+  }
+
+  /**
+   * Register this bucket as a required resource for the calling function/container.
```
Comment (davemooreuws) [inline src/resources/websocket.ts:79]: Change these comments

File: src/resources/websocket.ts [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+  async url(): Promise<string> {
+    const {
+      details: { url },
+    } = await this.details();
+
+    return url;
+  }
+
+  /**
+   * Register and start a bucket notification handler that will be called for all matching notification events on this bucket
```
Comment (davemooreuws) [inline src/resources/websocket.ts:153]: Change these comments

**Review by davemooreuws — CHANGES_REQUESTED** [review]
LGTM just a couple of comment changes

File: src/api/websocket/v0/websocket.ts (hunk 1 above)
Comment (tjholm) [reply src/api/websocket/v0/websocket.ts]: removed example as not our intended usage

File: src/api/websocket/v0/websocket.ts (hunk 2 above)
Comment (tjholm) [reply src/api/websocket/v0/websocket.ts]: done

**Review by jyecusch — APPROVED** [review]

**Review by davemooreuws — APPROVED** [review]

**Pull request merged**